### ISSUE TITLE
Error when compiling with SMCP_EMBEDDED set to true

### ISSUE THREAD

**Issue by blauskaerm** [darconeous/smcp#53]
Copied from Google-code form, see below:

Hi guys,

Have some problems compiling when SMCP_EMBEDDED is set to true. I saw on github that there seems to be some problems using the embedded flag with the new libraries. See https://github.com/darconeous/smcp/issues/32

Has there been any development with this issue? Would really want to test your stack on a TI CC3200 wifi chip since it seems to be a very competent lib.

I'm using the master branch and cloned the lib from github about a month ago. So it should be fairly new (checkout 3ee78de).

I have modified the following parameters in smcp-opts.h

```
#define SMCP_EMBEDDED          (1)
#define SMCP_USE_BSD_SOCKETS    1
#define SMCP_USE_UIP            0
```

CC3200 has its own ip stack which is more or less the same as the BSD socket library. So was hoping that this configure would work together.

Here is the output I get from make:

```
Making all in src
make[1]: Entering directory '/home/phoenix/Desktop/Coap/smcp/src'
make  all-recursive
make[2]: Entering directory '/home/phoenix/Desktop/Coap/smcp/src'
Making all in smcp
make[3]: Entering directory '/home/phoenix/Desktop/Coap/smcp/src/smcp'
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp.lo -MD -MP -MF .deps/libsmcp_la-smcp.Tpo -c -o libsmcp_la-smcp.lo `test -f 'smcp.c' || echo './'`smcp.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp.lo -MD -MP -MF .deps/libsmcp_la-smcp.Tpo -c smcp.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp.lo -MD -MP -MF .deps/libsmcp_la-smcp.Tpo -c smcp.c -o libsmcp_la-smcp.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp.Tpo .deps/libsmcp_la-smcp.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-timer.lo -MD -MP -MF .deps/libsmcp_la-smcp-timer.Tpo -c -o libsmcp_la-smcp-timer.lo `test -f 'smcp-timer.c' || echo './'`smcp-timer.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-timer.lo -MD -MP -MF .deps/libsmcp_la-smcp-timer.Tpo -c smcp-timer.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-timer.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-timer.lo -MD -MP -MF .deps/libsmcp_la-smcp-timer.Tpo -c smcp-timer.c -o libsmcp_la-smcp-timer.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-timer.Tpo .deps/libsmcp_la-smcp-timer.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-coap.lo -MD -MP -MF .deps/libsmcp_la-coap.Tpo -c -o libsmcp_la-coap.lo `test -f 'coap.c' || echo './'`coap.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-coap.lo -MD -MP -MF .deps/libsmcp_la-coap.Tpo -c coap.c  -fPIC -DPIC -o .libs/libsmcp_la-coap.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-coap.lo -MD -MP -MF .deps/libsmcp_la-coap.Tpo -c coap.c -o libsmcp_la-coap.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-coap.Tpo .deps/libsmcp_la-coap.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-outbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-outbound.Tpo -c -o libsmcp_la-smcp-outbound.lo `test -f 'smcp-outbound.c' || echo './'`smcp-outbound.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-outbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-outbound.Tpo -c smcp-outbound.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-outbound.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-outbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-outbound.Tpo -c smcp-outbound.c -o libsmcp_la-smcp-outbound.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-outbound.Tpo .deps/libsmcp_la-smcp-outbound.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-inbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-inbound.Tpo -c -o libsmcp_la-smcp-inbound.lo `test -f 'smcp-inbound.c' || echo './'`smcp-inbound.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-inbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-inbound.Tpo -c smcp-inbound.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-inbound.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-inbound.lo -MD -MP -MF .deps/libsmcp_la-smcp-inbound.Tpo -c smcp-inbound.c -o libsmcp_la-smcp-inbound.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-inbound.Tpo .deps/libsmcp_la-smcp-inbound.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-observable.lo -MD -MP -MF .deps/libsmcp_la-smcp-observable.Tpo -c -o libsmcp_la-smcp-observable.lo `test -f 'smcp-observable.c' || echo './'`smcp-observable.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-observable.lo -MD -MP -MF .deps/libsmcp_la-smcp-observable.Tpo -c smcp-observable.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-observable.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-observable.lo -MD -MP -MF .deps/libsmcp_la-smcp-observable.Tpo -c smcp-observable.c -o libsmcp_la-smcp-observable.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-observable.Tpo .deps/libsmcp_la-smcp-observable.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-auth.lo -MD -MP -MF .deps/libsmcp_la-smcp-auth.Tpo -c -o libsmcp_la-smcp-auth.lo `test -f 'smcp-auth.c' || echo './'`smcp-auth.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-auth.lo -MD -MP -MF .deps/libsmcp_la-smcp-auth.Tpo -c smcp-auth.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-auth.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-auth.lo -MD -MP -MF .deps/libsmcp_la-smcp-auth.Tpo -c smcp-auth.c -o libsmcp_la-smcp-auth.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-auth.Tpo .deps/libsmcp_la-smcp-auth.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-transaction.lo -MD -MP -MF .deps/libsmcp_la-smcp-transaction.Tpo -c -o libsmcp_la-smcp-transaction.lo `test -f 'smcp-transaction.c' || echo './'`smcp-transaction.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-transaction.lo -MD -MP -MF .deps/libsmcp_la-smcp-transaction.Tpo -c smcp-transaction.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-transaction.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-transaction.lo -MD -MP -MF .deps/libsmcp_la-smcp-transaction.Tpo -c smcp-transaction.c -o libsmcp_la-smcp-transaction.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-transaction.Tpo .deps/libsmcp_la-smcp-transaction.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-dupe.lo -MD -MP -MF .deps/libsmcp_la-smcp-dupe.Tpo -c -o libsmcp_la-smcp-dupe.lo `test -f 'smcp-dupe.c' || echo './'`smcp-dupe.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-dupe.lo -MD -MP -MF .deps/libsmcp_la-smcp-dupe.Tpo -c smcp-dupe.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-dupe.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-dupe.lo -MD -MP -MF .deps/libsmcp_la-smcp-dupe.Tpo -c smcp-dupe.c -o libsmcp_la-smcp-dupe.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-dupe.Tpo .deps/libsmcp_la-smcp-dupe.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-missing.lo -MD -MP -MF .deps/libsmcp_la-smcp-missing.Tpo -c -o libsmcp_la-smcp-missing.lo `test -f 'smcp-missing.c' || echo './'`smcp-missing.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-missing.lo -MD -MP -MF .deps/libsmcp_la-smcp-missing.Tpo -c smcp-missing.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-missing.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-missing.lo -MD -MP -MF .deps/libsmcp_la-smcp-missing.Tpo -c smcp-missing.c -o libsmcp_la-smcp-missing.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-missing.Tpo .deps/libsmcp_la-smcp-missing.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-plat-bsd.lo -MD -MP -MF .deps/libsmcp_la-smcp-plat-bsd.Tpo -c -o libsmcp_la-smcp-plat-bsd.lo `test -f 'smcp-plat-bsd.c' || echo './'`smcp-plat-bsd.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-plat-bsd.lo -MD -MP -MF .deps/libsmcp_la-smcp-plat-bsd.Tpo -c smcp-plat-bsd.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-plat-bsd.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-plat-bsd.lo -MD -MP -MF .deps/libsmcp_la-smcp-plat-bsd.Tpo -c smcp-plat-bsd.c -o libsmcp_la-smcp-plat-bsd.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-plat-bsd.Tpo .deps/libsmcp_la-smcp-plat-bsd.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-btree.lo -MD -MP -MF .deps/libsmcp_la-btree.Tpo -c -o libsmcp_la-btree.lo `test -f 'btree.c' || echo './'`btree.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-btree.lo -MD -MP -MF .deps/libsmcp_la-btree.Tpo -c btree.c  -fPIC -DPIC -o .libs/libsmcp_la-btree.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-btree.lo -MD -MP -MF .deps/libsmcp_la-btree.Tpo -c btree.c -o libsmcp_la-btree.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-btree.Tpo .deps/libsmcp_la-btree.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-url-helpers.lo -MD -MP -MF .deps/libsmcp_la-url-helpers.Tpo -c -o libsmcp_la-url-helpers.lo `test -f 'url-helpers.c' || echo './'`url-helpers.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-url-helpers.lo -MD -MP -MF .deps/libsmcp_la-url-helpers.Tpo -c url-helpers.c  -fPIC -DPIC -o .libs/libsmcp_la-url-helpers.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-url-helpers.lo -MD -MP -MF .deps/libsmcp_la-url-helpers.Tpo -c url-helpers.c -o libsmcp_la-url-helpers.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-url-helpers.Tpo .deps/libsmcp_la-url-helpers.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-fasthash.lo -MD -MP -MF .deps/libsmcp_la-fasthash.Tpo -c -o libsmcp_la-fasthash.lo `test -f 'fasthash.c' || echo './'`fasthash.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-fasthash.lo -MD -MP -MF .deps/libsmcp_la-fasthash.Tpo -c fasthash.c  -fPIC -DPIC -o .libs/libsmcp_la-fasthash.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-fasthash.lo -MD -MP -MF .deps/libsmcp_la-fasthash.Tpo -c fasthash.c -o libsmcp_la-fasthash.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-fasthash.Tpo .deps/libsmcp_la-fasthash.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-string-utils.lo -MD -MP -MF .deps/libsmcp_la-string-utils.Tpo -c -o libsmcp_la-string-utils.lo `test -f 'string-utils.c' || echo './'`string-utils.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-string-utils.lo -MD -MP -MF .deps/libsmcp_la-string-utils.Tpo -c string-utils.c  -fPIC -DPIC -o .libs/libsmcp_la-string-utils.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-string-utils.lo -MD -MP -MF .deps/libsmcp_la-string-utils.Tpo -c string-utils.c -o libsmcp_la-string-utils.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-string-utils.Tpo .deps/libsmcp_la-string-utils.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-node-router.lo -MD -MP -MF .deps/libsmcp_la-smcp-node-router.Tpo -c -o libsmcp_la-smcp-node-router.lo `test -f 'smcp-node-router.c' || echo './'`smcp-node-router.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-node-router.lo -MD -MP -MF .deps/libsmcp_la-smcp-node-router.Tpo -c smcp-node-router.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-node-router.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-node-router.lo -MD -MP -MF .deps/libsmcp_la-smcp-node-router.Tpo -c smcp-node-router.c -o libsmcp_la-smcp-node-router.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-node-router.Tpo .deps/libsmcp_la-smcp-node-router.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-list.lo -MD -MP -MF .deps/libsmcp_la-smcp-list.Tpo -c -o libsmcp_la-smcp-list.lo `test -f 'smcp-list.c' || echo './'`smcp-list.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-list.lo -MD -MP -MF .deps/libsmcp_la-smcp-list.Tpo -c smcp-list.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-list.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-list.lo -MD -MP -MF .deps/libsmcp_la-smcp-list.Tpo -c smcp-list.c -o libsmcp_la-smcp-list.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-list.Tpo .deps/libsmcp_la-smcp-list.Plo
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -pthread -g -O2 -MT libsmcp_la-smcp-variable_node.lo -MD -MP -MF .deps/libsmcp_la-smcp-variable_node.Tpo -c -o libsmcp_la-smcp-variable_node.lo `test -f 'smcp-variable_node.c' || echo './'`smcp-variable_node.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-variable_node.lo -MD -MP -MF .deps/libsmcp_la-smcp-variable_node.Tpo -c smcp-variable_node.c  -fPIC -DPIC -o .libs/libsmcp_la-smcp-variable_node.o
smcp-variable_node.c: In function 'smcp_variable_node_request_handler':
smcp-variable_node.c:305:9: warning: implicit declaration of function 'smcp_default_request_handler' [-Wimplicit-function-declaration]
   ret = smcp_default_request_handler(
         ^
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src -pthread -g -O2 -MT libsmcp_la-smcp-variable_node.lo -MD -MP -MF .deps/libsmcp_la-smcp-variable_node.Tpo -c smcp-variable_node.c -o libsmcp_la-smcp-variable_node.o >/dev/null 2>&1
mv -f .deps/libsmcp_la-smcp-variable_node.Tpo .deps/libsmcp_la-smcp-variable_node.Plo
/bin/bash ../../libtool  --tag=CC   --mode=link gcc -pthread -g -O2   -o libsmcp.la -rpath /usr/local/lib libsmcp_la-smcp.lo libsmcp_la-smcp-timer.lo libsmcp_la-coap.lo libsmcp_la-smcp-outbound.lo libsmcp_la-smcp-inbound.lo libsmcp_la-smcp-observable.lo libsmcp_la-smcp-auth.lo libsmcp_la-smcp-transaction.lo libsmcp_la-smcp-dupe.lo libsmcp_la-smcp-missing.lo libsmcp_la-smcp-plat-bsd.lo libsmcp_la-btree.lo libsmcp_la-url-helpers.lo libsmcp_la-fasthash.lo libsmcp_la-string-utils.lo libsmcp_la-smcp-node-router.lo libsmcp_la-smcp-list.lo libsmcp_la-smcp-variable_node.lo     -ldl -lreadline
libtool: link: gcc -shared  -fPIC -DPIC  .libs/libsmcp_la-smcp.o .libs/libsmcp_la-smcp-timer.o .libs/libsmcp_la-coap.o .libs/libsmcp_la-smcp-outbound.o .libs/libsmcp_la-smcp-inbound.o .libs/libsmcp_la-smcp-observable.o .libs/libsmcp_la-smcp-auth.o .libs/libsmcp_la-smcp-transaction.o .libs/libsmcp_la-smcp-dupe.o .libs/libsmcp_la-smcp-missing.o .libs/libsmcp_la-smcp-plat-bsd.o .libs/libsmcp_la-btree.o .libs/libsmcp_la-url-helpers.o .libs/libsmcp_la-fasthash.o .libs/libsmcp_la-string-utils.o .libs/libsmcp_la-smcp-node-router.o .libs/libsmcp_la-smcp-list.o .libs/libsmcp_la-smcp-variable_node.o   -ldl -lreadline  -pthread -O2   -pthread -Wl,-soname -Wl,libsmcp.so.0 -o .libs/libsmcp.so.0.0.0
libtool: link: (cd ".libs" && rm -f "libsmcp.so.0" && ln -s "libsmcp.so.0.0.0" "libsmcp.so.0")
libtool: link: (cd ".libs" && rm -f "libsmcp.so" && ln -s "libsmcp.so.0.0.0" "libsmcp.so")
libtool: link: ar cru .libs/libsmcp.a  libsmcp_la-smcp.o libsmcp_la-smcp-timer.o libsmcp_la-coap.o libsmcp_la-smcp-outbound.o libsmcp_la-smcp-inbound.o libsmcp_la-smcp-observable.o libsmcp_la-smcp-auth.o libsmcp_la-smcp-transaction.o libsmcp_la-smcp-dupe.o libsmcp_la-smcp-missing.o libsmcp_la-smcp-plat-bsd.o libsmcp_la-btree.o libsmcp_la-url-helpers.o libsmcp_la-fasthash.o libsmcp_la-string-utils.o libsmcp_la-smcp-node-router.o libsmcp_la-smcp-list.o libsmcp_la-smcp-variable_node.o
ar: `u' modifier ignored since `D' is the default (see `U')
libtool: link: ranlib .libs/libsmcp.a
libtool: link: ( cd ".libs" && rm -f "libsmcp.la" && ln -s "../libsmcp.la" "libsmcp.la" )
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src   -DBTREE_SELF_TEST=1 -g -O2 -MT btreetest-btree.o -MD -MP -MF .deps/btreetest-btree.Tpo -c -o btreetest-btree.o `test -f 'btree.c' || echo './'`btree.c
btree.c: In function ‘main’:
btree.c:531:3: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   asprintf(&new_node->name, "item_%d", c);
   ^
mv -f .deps/btreetest-btree.Tpo .deps/btreetest-btree.Po
/bin/bash ../../libtool  --tag=CC   --mode=link gcc -DBTREE_SELF_TEST=1 -g -O2   -o btreetest btreetest-btree.o  -ldl -lreadline
libtool: link: gcc -DBTREE_SELF_TEST=1 -g -O2 -o btreetest btreetest-btree.o  -ldl -lreadline
make[3]: Leaving directory '/home/phoenix/Desktop/Coap/smcp/src/smcp'
Making all in smcpctl
make[3]: Entering directory '/home/phoenix/Desktop/Coap/smcp/src/smcpctl'
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT main.o -MD -MP -MF .deps/main.Tpo -c -o main.o main.c
main.c: In function ‘smcp_directory_generator’:
main.c:448:5: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
     asprintf(&cmdline, "list --filename-only --timeout 750 /.well-known/core");
     ^
main.c: In function ‘smcp_attempted_completion’:
main.c:537:13: warning: implicit declaration of function ‘completion_matches’ [-Wimplicit-function-declaration]
   matches = completion_matches (text, &smcp_command_generator);
             ^
main.c:537:11: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   matches = completion_matches (text, &smcp_command_generator);
           ^
main.c: In function ‘main’:
main.c:737:4: warning: ignoring return value of ‘fgets’, declared with attribute warn_unused_result [-Wunused-result]
    fgets(linebuffer, sizeof(linebuffer), stdin);
    ^
mv -f .deps/main.Tpo .deps/main.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT cmd_list.o -MD -MP -MF .deps/cmd_list.Tpo -c -o cmd_list.o cmd_list.c
mv -f .deps/cmd_list.Tpo .deps/cmd_list.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT cmd_get.o -MD -MP -MF .deps/cmd_get.Tpo -c -o cmd_get.o cmd_get.c
mv -f .deps/cmd_get.Tpo .deps/cmd_get.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT cmd_post.o -MD -MP -MF .deps/cmd_post.Tpo -c -o cmd_post.o cmd_post.c
mv -f .deps/cmd_post.Tpo .deps/cmd_post.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT help.o -MD -MP -MF .deps/help.Tpo -c -o help.o help.c
mv -f .deps/help.Tpo .deps/help.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT cmd_repeat.o -MD -MP -MF .deps/cmd_repeat.Tpo -c -o cmd_repeat.o cmd_repeat.c
mv -f .deps/cmd_repeat.Tpo .deps/cmd_repeat.Po
gcc -DHAVE_CONFIG_H -I. -I../../src -I.. -I../../src    -g -O2 -MT cmd_delete.o -MD -MP -MF .deps/cmd_delete.Tpo -c -o cmd_delete.o cmd_delete.c
mv -f .deps/cmd_delete.Tpo .deps/cmd_delete.Po
/bin/bash ../../libtool  --tag=CC   --mode=link gcc  -g -O2   -o smcpctl main.o cmd_list.o cmd_get.o cmd_post.o help.o cmd_repeat.o cmd_delete.o ../smcp/libsmcp.la -ldl -lreadline
libtool: link: gcc -g -O2 -o .libs/smcpctl main.o cmd_list.o cmd_get.o cmd_post.o help.o cmd_repeat.o cmd_delete.o  ../smcp/.libs/libsmcp.so -ldl -lreadline -pthread
cmd_list.o: In function `resend_list_request':
/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_list.c:373: undefined reference to `smcp_status_to_cstr'
cmd_list.o: In function `list_response_handler':
/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_list.c:290: undefined reference to `smcp_status_to_cstr'
cmd_list.o: In function `send_list_request':
/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_list.c:422: undefined reference to `smcp_status_to_cstr'
cmd_get.o: In function `get_response_handler':
/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_get.c:120: undefined reference to `smcp_status_to_cstr'
cmd_get.o: In function `resend_get_request':
/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_get.c:186: undefined reference to `smcp_status_to_cstr'
cmd_get.o:/home/phoenix/Desktop/Coap/smcp/src/smcpctl/cmd_get.c:226: more undefined references to `smcp_status_to_cstr' follow
../smcp/.libs/libsmcp.so: undefined reference to `smcp_default_request_handler'
collect2: error: ld returned 1 exit status
Makefile:379: recipe for target 'smcpctl' failed
make[3]: *** [smcpctl] Error 1
make[3]: Leaving directory '/home/phoenix/Desktop/Coap/smcp/src/smcpctl'
Makefile:362: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/phoenix/Desktop/Coap/smcp/src'
Makefile:300: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/phoenix/Desktop/Coap/smcp/src'
Makefile:397: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by blauskaerm** [comment]
I did some digging in the first declaration

`smcp-variable_node.c:305:9: warning: implicit declaration of function 'smcp_default_request_handler' [-Wimplicit-function-declaration]`

There seems to be a conflict in smcp-variable_node.c when including smcp-node-router.h. The header content seems to be guarded by SMCP_CONF_NODE_ROUTER when is set to the inverted value of SMCP_EMBEDDED in smcp-opts.h

**Comment by darconeous** [comment]
Is that really the make output from a cross-compile? 

The combination of `SMCP_EMBEDDED` and `SMCP_USE_BSD_SOCKETS` isn't super well tested. 

But in your case it actually looks like the library successfully built:

    ....
    libtool: link: ranlib .libs/libsmcp.a
    libtool: link: ( cd ".libs" && rm -f "libsmcp.la" && ln -s "../libsmcp.la" "libsmcp.la" )
    ....

You should be able to `cd` into the `src/smcp` directory and do a `make install DESTDIR=<dest-root-dir>` successfully.

The errors you are seeing are because `smcpctl` depends on some methods that only exist when `SMCP_EMBEDDED` is turned off. But you shouldn't care too much about that because you won't be running `smcpctl` on your embedded device. :)

Nonetheless, the build system in SMCP needs an overhaul to prevent things like `smcpctl` from building if the library doesn't have the features needed to support it.

I plan to get back to this project pretty soon because my house construction is almost complete and I'll need to get this software working on all my light switches. :)

**Comment by darconeous** [comment]
Does the CC3200 use LWIP? At some point I want to implement native LWIP support.

**Comment by darconeous** [comment]
I believe this should be fixed by #54.

**Comment by darconeous** [comment]
I also don't see any reference to `smcp_default_request_handler()` in the current version of `smcp-variable_node.c`.
